clang/test/Preprocessor/sycl-int-footer.cpp

```diff
@@ -0,0 +1,48 @@
+// This test verifies that the integration footer is appended to the source
+// file.
+
+// RUN: %clangxx  -E -fsycl %S/Inputs/checkfooter.cpp \
+// RUN: -Xclang -include -Xclang %S/Inputs/file1.h \
+// RUN: -Xclang -include-footer -Xclang %S/Inputs/file2.h \
+// RUN: | FileCheck %s
+
+// CHECK: // __CLANG_OFFLOAD_BUNDLE____START__ sycl-spir64
+// CHECK: # 1 "[[INPUTFILE:.+\.cpp]]"
+// CHECK: # 1 "[[INTHEADER:.+\.h]]" 1
+// CHECK: int file1() {
+// CHECK:   return 1;
+// CHECK: }
+// CHECK: # 2 "<built-in>" 2
+// CHECK: # 1 "[[INPUTFILE:.+\.cpp]]" 2
+// CHECK: int main() {
+// CHECK:   int i = 0;
+// CHECK:   return i++;
+// CHECK: }
+// CHECK: # 1 "[[INTFOOTER:.+\.h]]" 1
+// CHECK: int file2() {
+// CHECK:   return 2;
+// CHECK: }
```

Comment:
Is the header file and footer file included in device compilations too now?

Reply:
The integration header and integration footer are only used in the host compilation.  Device compilations are when the header/footer are created.

Reply:
I can see where the confusion is - adding `-fsycl` here implies we are adding the header and footer to both the host and device compilations.  I think just a regular host compilation without `-fsycl` to test should be sufficient.

Reply:
Since the `-include-footer` will be used only in host compilation, I have updated the LIT test to reflect that.